### PR TITLE
fix(tracing): add span processors shutdown

### PR DESCRIPTION
**Pull request by mabdinur**
Spans should not be sent to the agent after ``Tracer.shutown()`` is called

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: releasenotes/notes/reset_span_processors_on_shutdown-3d63f33e1354cd5e.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Spans should not be sent to the Agent after py:meth:`ddtrace.Tracer.shutdown` is called.
```
Comment (majorgreys) [inline releasenotes/notes/reset_span_processors_on_shutdown-3d63f33e1354cd5e.yaml:4]:
```suggestion
    Tracer writer service is no longer restarted after the tracer is shutdown.
```

File: ddtrace/tracer.py [hunk 2]
```diff
@@ -901,6 +901,7 @@ def shutdown(self, timeout=None):
         :type timeout: :obj:`int` | :obj:`float` | :obj:`None`
         """
         try:
+            self._span_processors = []
             self._writer.stop(timeout=timeout)
```
Comment (Kyle-Verhoog) [inline ddtrace/tracer.py:905]: Can move this to `SpanAggregator.shutdown` (resolve that TODO)

File: ddtrace/tracer.py [hunk 3]
```diff
@@ -901,6 +901,7 @@ def shutdown(self, timeout=None):
         :type timeout: :obj:`int` | :obj:`float` | :obj:`None`
         """
         try:
+            self._span_processors = []
```
Comment (Kyle-Verhoog) [inline ddtrace/tracer.py:904]:
```
        for proc in self._span_processors:
            if hasattr(proc, "shutdown"):
                proc.shutdown()
```

and then clear the list

File: ddtrace/tracer.py [hunk 4]
```diff
@@ -893,18 +893,16 @@ def shutdown(self, timeout=None):
         # type: (Optional[float]) -> None
         """Shutdown the tracer.
 
-        This will stop the background writer/worker and flush any finished traces in the buffer. The tracer cannot be
-        used for tracing after this method has been called. A new tracer instance is required to continue tracing.
+        This will call shutdown on all SpanProcessors and stop the AgentWriter.
```
Comment (Kyle-Verhoog) [inline ddtrace/tracer.py:896]:
```suggestion
```

`SpanProcessor`s are not a public API (yet) so we shouldn't be mentioning them in the public docstring

**Review by majorgreys — DISMISSED** [review]

**Comment by majorgreys** [comment]
@Mergifyio backport 1.0

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Pull request merged**